### PR TITLE
OCPBUGS-33567: Expose useUserSettings hooks in dynamic plugin sdk

### PR DESCRIPTION
**Pull request by rhamilto**

### CODE REVIEW

File: frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts [hunk 1]
```diff
@@ -877,3 +878,23 @@ export const useLabelsModal: UseLabelsModal = require('@console/shared/src/hooks
...
+ *      true,
+ *    );
+ *    return loaded ? (
+ *       <WrappedComponent {...props} userSettingState={state} setUserSettingState={setState} />
+ *     ) : null;
+ * };
+ * ```
+ */
+export const useUserSettings: UseUserSettings = require('@console/shared/src/hooks/useUserSettings')
+.useUserSettings;
```
Comment (vojtechszocs) [inline frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts:900]: Nit: formatting
```suggestion
  .useUserSettings;
```

File: frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts [hunk 2]
```diff
@@ -734,6 +734,12 @@ export type UseValuesForNamespaceContext = () => {
 
 export type UseActiveNamespace = () => [string, (ns: string) => void];
 
+export type UseUserSettings = <T>(
```
Comment (vojtechszocs) [inline frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts:737]: This type is also defined in `packages/console-dynamic-plugin-sdk/src/api/internal-types.ts`

Can we remove that one in favor of this one?

Comment (rhamilto) [reply frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts:737]: Good catch.  That was an oversight.  I thought I'd removed it.

**Pull request merged**